src/app/pages/viewing/single-message/single-message.component.ts

```diff
@@ -0,0 +1,32 @@
+import { Component, OnInit } from '@angular/core';
+import { ViewingService } from '../viewing.service';
+import { ActivatedRoute, Router } from '@angular/router';
+
+@Component({
+  selector: 'app-single-message',
+  templateUrl: './single-message.component.html',
+  styleUrls: [ './single-message.component.css' ]
+})
+export class SingleMessageComponent implements OnInit {
+  currentMessageId: string;
+
+  constructor(public viewService: ViewingService, private route: ActivatedRoute, private router: Router) {
+    this.route.queryParams.subscribe(params => {
+      this.currentMessageId = params.messageId;
+    });
+  }
+
+  ngOnInit(): void {
+    this.viewService.getMessageById(this.currentMessageId);
+  }
+
+  /**
+   * Go to 'all messages' view
+   */
+  goToMessages() {
+    // noinspection JSUnusedLocalSymbols
+    this.router.navigate([ '/viewing' ]).then(navigate => {
```

Comment:
Removed unused local symbol
```suggestion
    this.router.navigate([ '/viewing' ]).then(() => {
```